contracts/MechMarketplace.sol

```diff
@@ -58,19 +57,25 @@ contract MechMarketplace is IErrorsMarketplace {
     // Maximum response time
     uint256 public immutable maxResponseTimeout;
     // Mech karma contract address
-    address public immutable karmaProxy;
+    address public immutable karma;
+    // Mech manager contract address
+    address public immutable mechManager;
     // Staking factory contract address
     address public immutable stakingFactory;
     // Service registry contract address
     address public immutable serviceRegistry;
 
+    // Collected fees
+    uint256 public collectedFees;
     // Number of undelivered requests
     uint256 public numUndeliveredRequests;
     // Number of total requests
     uint256 public numTotalRequests;
     // Reentrancy lock
     uint256 internal _locked = 1;
 
+    // Map of request payments
+    mapping(uint256 => uint256) public mapRequestIdPayments;
```

Comment:
Why do we need this? Can we not get away with a fee counter per mech?

Reply:
I don't understand what is meant by that. Funds don't go to mechs before requests are delivered, but payment is taken during the request. What is your suggestion?